samples/GalleryApp/ObjectDetectorPage.xaml.cs

```diff
@@ -0,0 +1,494 @@
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace GalleryApp
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class ObjectDetectorPage : Page
+    {
+        private IFrameSource m_frameSource = null;
+
+        // Vision Skills
+        private ObjectDetectorDescriptor m_descriptor = null;
+        private ObjectDetectorBinding m_binding = null;
+        private ObjectDetectorSkill m_skill = null;
+        private IReadOnlyList<ISkillExecutionDevice> m_availableExecutionDevices = null;
+
+        // Misc
+        private BoundingBoxRenderer m_bboxRenderer = null;
+        private HashSet<ObjectKind> m_objectKinds = null;
+
+        // Frames
+        private SoftwareBitmapSource m_processedBitmapSource;
+
+        // Performance metrics
+        private Stopwatch m_evalStopwatch = new Stopwatch();
+        private float m_bindTime = 0;
+        private float m_evalTime = 0;
+        private Stopwatch m_renderStopwatch = new Stopwatch();
+
+        // Locks
+        private SemaphoreSlim m_lock = new SemaphoreSlim(1);
+
+        public ObjectDetectorPage()
+        {
+            this.InitializeComponent();
+        }
+
+        /// <summary>
+        /// Backward navigation to MainPage
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Back_Click(object sender, RoutedEventArgs e)
+        {
+            this.Frame.Navigate(typeof(MainPage));
+        }
+
+        /// <summary>
+        /// Called when page is loaded
+        /// Initialize app assets such as skills
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void Page_Loaded(object sender, RoutedEventArgs e)
+        {
+            // Disable buttons while we initialize
+            await UpdateMediaSourceButtonsAsync(false);
+
+            // Initialize helper class used to render the skill results on screen
+            m_bboxRenderer = new BoundingBoxRenderer(UIOverlayCanvas);
+
+            m_lock.Wait();
+            {
+                NotifyUser("Initializing skill...");
+                m_descriptor = new ObjectDetectorDescriptor();
+                m_availableExecutionDevices = await m_descriptor.GetSupportedExecutionDevicesAsync();
+
+                await InitializeObjectDetectorAsync();
+                await UpdateSkillUIAsync();
+            }
+            m_lock.Release();
+
+            // Ready to begin, enable buttons
+            NotifyUser("Skill initialized. Select a media source from the top to begin.");
+            await UpdateMediaSourceButtonsAsync(true);
+        }
+
+        /// <summary>
+        /// Populate UI with skill information and options
+        /// </summary>
+        /// <returns></returns>
+        private async Task UpdateSkillUIAsync()
```

Comment:
suggestion: consider writing this only once in a helper method accessible to all skills

Reply:
Task 22491451: Refactor codes - UpdateSkillUIAsync

---
In reply to: [299505188](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/39#discussion_r299505188) [](ancestors = 299505188)